nats.go

```diff
@@ -1783,6 +1784,21 @@ func (nc *Conn) ConnectedServerName() string {
 	return nc.info.Name
 }
 
+// ConnectedServerVersion reports the connected server's version
+func (nc *Conn) ConnectedServerVersion() string {
```

Comment:
Feels like we need a better name.

Reply:
Goes with `ConnectedServerId` and `ConnectedServerName`

Reply:
Oh geez we have those? Must have missed that :/

Reply:
We should move to just Server() and return and interface or struct that has ID, Name, Node, Version etc..